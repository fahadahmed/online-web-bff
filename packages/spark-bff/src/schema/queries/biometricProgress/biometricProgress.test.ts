import { createTestClient, gql } from 'test/apolloTestUtils';

test('biometricProgress query responds with correct mocked data', async () => {
  const { query } = createTestClient();

  const res = await query({
    query: gql`
      query BiometricProgressQuery($cartId: String!, $channel: CartChannel!) {
        biometricProgress(cartId: $cartId, channel: $channel) {
          infoBanner
          isContinueEnabled
          showResend
          steps {
            stepId
            label
            status
          }
        }
      }
    `,
    variables: {
      cartId: 'success-cart-id',
      channel: 'personalshop',
    },
  });
  expect(res.errors).toBeUndefined();
  expect(res.data).toMatchSnapshot();
});
