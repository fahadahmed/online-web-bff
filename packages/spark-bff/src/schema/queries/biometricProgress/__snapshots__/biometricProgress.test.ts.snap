// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`biometricProgress query responds with correct mocked data 1`] = `
Object {
  "biometricProgress": Object {
    "infoBanner": "NO_MATCH",
    "isContinueEnabled": false,
    "showResend": true,
    "steps": Array [
      Object {
        "label": "Text sent",
        "status": "DONE",
        "stepId": "TEXT_SENT",
      },
      Object {
        "label": "Results",
        "status": "ERROR",
        "stepId": "RESULTS",
      },
    ],
  },
}
`;
